app/src/app/api/v0/emissions-factor/route.ts

```diff
@@ -104,16 +114,61 @@ export const POST = apiHandler(async (req: NextRequest, _context: {}) => {
       whereClause.methodologyName = "energy-consumption";
   }
 
+  // Unified priority list (order matters)
+  const priorityArray: string[] = [];
+
+  if (region_name) priorityArray.push(region_name);
+  if (region_locode) priorityArray.push(region_locode);
+  if (city_name) priorityArray.push(city_name);
+  if (country) priorityArray.push(country);
+  if (countryLocode) priorityArray.push(countryLocode);
+  priorityArray.push("world"); // fallback for region
+
+  whereClause = {
+    ...whereClause,
+    [Op.or]: [
+      { region: { [Op.in]: priorityArray } },
+      { actorId: { [Op.in]: priorityArray } },
+    ],
+  };
+
+  const caseStatements = priorityArray
+    .map((value, index) => {
+      const rank = index + 1;
+      return `
+      WHEN "EmissionsFactor"."region" = '${value}' THEN ${rank}
+      WHEN "EmissionsFactor"."actor_id" = '${value}' THEN ${rank}
```

Comment:
Don't put raw values (especially when they are coming from the user) into database queries. Sequelize has replacements and bind parameters for raw queries which you can use to safely inject the values into the query (without risking SQL injection): https://sequelize.org/docs/v6/core-concepts/raw-queries/#replacements